### DISCUSSION TITLE
propose a common protocol  - discovery

### DISCUSSION THREAD

**Discussion by marcellanz** [eigr/massa#59] [Ideas]
The text below would be a proposal sent to the Cloudstate project:

Hi there👋

Cloudstate is in limbo and not much is going on with the project. At [eigr.io](https://eigr.io) originaly we're based on the protocol Cloudstate is based on, but we will not stay compatible, especially because of Cloudstates current condition as an open source project.

At the moment, eigr/massa is probably the most compatible implementation of the Cloudstate protocol, beside the reference implementation itself. Akka Serverless seems to be incompatible with Cloudstate.

In an effort to support open protocols in the space of serverless runtimes, we propose to find and agree on a common protocol definition that enables runtimes to discover if a deployed service is compatible. In the case of the current three projects that are based on Cloudstate, with Cloudstate itself, we propose to agree on a common protocol definition for the discovery service to start with.

We hope, that with this step, the legacy of Cloudstate, even if not evolved itself, can be preserved as a reference and partly a spec that others can build on. We're aware and expect that implementations like Akka Serverless and eigr.io are and will diverge further in terms of their entities types and implementation compatibilty. Saying this, we would welcome if some of the entity types with their state- or action-models can be compatible in the future. This proposal would be a common ground for such a compatibility.

At the moment, we don't think staying compatible with Cloudstate enables us to go forward with eigr.io, so we pursue evolving the protocol. If we find common ground on this proposal, we hope different serverless runtimes based on Cloudstate in the future can coexist and even collaborate in their common believes what serverless can be and where it can evolve to.

Yours sincerely, the team at eigr.io

**Comment by sleipnir** [comment]
> The text below would be a proposal sent to the Cloudstate project:
> ......
> At the moment, we don't think being compatible with cloudstate enables us to go forward with eigr.io, so we pursue evolving the protocol. If we find common ground on this proposal, we hope different serverless runtimes based on Cloudstate in the future can coexist and even collaborate in their common believes what serverless can be and where it can evolve to.
> ....

I thought it was perfect.
I would make my technical comments following this text when I open the Issue in the Cloudstate repository.
I agree to ping Viktor, Jonas and Peter

**Reply by marcellanz** [reply]
cool, updated a bit for formatting and words. see edit history.
